### PR TITLE
Ticket 23: How can TLS clients match an SCT to a certificate?

### PR DESCRIPTION
**Pull request by robstradling**

### CODE REVIEW

File: rfc6962-bis.xml [hunk 1]
```diff
@@ -1320,24 +1298,38 @@ but it is expected there will be a variety.
...
+          </t>
+          <t>
+            TODO: What should the TLS client communicate in the extension_data? Version(s) of CT that it supports? Certain types of TransItem that it can handle? Whether or not it wants to gossip?
+          </t>
+        </section>
+        <section title="Validating SCTs">
+          <t>
+            In addition to normal validation of the server certificate and its chain, TLS clients SHOULD validate each received SCT by computing the signature input from the SCT data as well as the corresponding certificate and verifying the signature, using the corresponding log's public key. TLS clients MUST reject SCTs whose timestamp is in the future.
```
Comment (benlaurie) [inline rfc6962-bis.xml]: Obvs can only validate if you know the log...

File: rfc6962-bis.xml [hunk 2]
```diff
@@ -1320,24 +1298,38 @@ but it is expected there will be a variety.
...
+          </t>
+        </section>
+        <section title="Validating inclusion proofs">
+          <t>
+            TLS clients SHOULD also verify each received inclusion proof (see <xref target="verify_inclusion"/>), to audit the log and gain confidence that the certificate is logged. Since each received inclusion proof may be valid either for the server certificate itself or for a <xref target="name_constrained">suitable name-constrained intermediate</xref>, TLS clients MAY need to use trial and error to determine which certificate in the chain is the certificate to which a received inclusion proof corresponds.
+          </t>
+          <t>
+            If no inclusion proof was received, the TLS client MAY request one directly from the corresponding log using <spanx style="verb">get-proof-by-hash</spanx> (<xref target="get-proof-by-hash"/>) or <spanx style="verb">get-all-by-hash</spanx> (<xref target="get-all-by-hash"/>), and then verify it. Note that this will disclose to the log which TLS server the client has been communicating with.
```
Comment (benlaurie) [inline rfc6962-bis.xml]: Since we plan alternate means to get the inclusion proof (e.g. DNS), we should allow for those here?

**Comment by benlaurie** [comment]
Nice. Couple of small nits...

File: rfc6962-bis.xml (hunk 1 above)
Comment (robstradling) [inline rfc6962-bis.xml]: Addressed by https://github.com/robstradling/certificate-transparency-rfcs/commit/2e8f6f4c981de9d605a771fa2d4e773f71e8f5d9

Comment (robstradling) [inline rfc6962-bis.xml]: Also, you can only validate an inclusion proof if you know the log.  Addressed by https://github.com/robstradling/certificate-transparency-rfcs/commit/bd83233f73709cd56083c27e2a243ffd319ae14a

File: rfc6962-bis.xml (hunk 2 above)
Comment (robstradling) [inline rfc6962-bis.xml]: Sure.  Do you have an I-D that we can reference for acquiring inclusion proofs via DNS?  Or do you just want the text here to leave open the option of acquiring inclusion proofs by alternative means?

Comment (robstradling) [inline rfc6962-bis.xml]: Also, I'm not sure "...directly from the corresponding log" quite makes sense.  Corresponding to what?

Comment (benlaurie) [inline rfc6962-bis.xml]: a) I meant to leave the option open (we will be talking about DNS very soon, though).

b) "corresponding" is a little odd, tho I think its obvious what it means - could perhaps just drop that word?

Comment (robstradling) [inline rfc6962-bis.xml]: b) addressed by https://github.com/robstradling/certificate-transparency-rfcs/commit/6720ed2f23ff039be40a0eaae5e1377254546f7b

**Comment by benlaurie** [comment]
Apart from my one comment, LGTM...

**Comment by robstradling** [comment]
Merged at https://github.com/google/certificate-transparency-rfcs/commit/d71d7707a3eeb5707cf5048c3849b068e477038c

**Pull request closed without merging**